lisp/ess-r-mode.el

```diff
@@ -604,17 +714,12 @@ Executed in process buffer."
   (add-hook 'xref-backend-functions #'ess-r-xref-backend nil 'local)
 
   (if (fboundp 'ess-add-toolbar) (ess-add-toolbar))
+  ;; imenu is needed for `which-function'
+  (setq imenu-generic-expression ess-imenu-S-generic-expression)
   (when ess-imenu-use-S
-    (setq imenu-generic-expression ess-imenu-S-generic-expression)
     (imenu-add-to-menubar "Imenu-R"))
-
-  ;; useful for swankr/slime:
-  (setq-local beginning-of-defun-function
-              (lambda (&optional arg)
-                (skip-chars-backward " \t\n")
-                (ess-beginning-of-function 'no-error)))
-  (setq-local end-of-defun-function
-              'ess-end-of-function)
+  (setq-local beginning-of-defun-function #'ess-r-beginning-of-function)
```

Comment:
Let's call them `ess-r-beginning-of-defun` and `*-end-of-defun` as per emacs standards.

Reply:
If we do that, should we also remove the interactive spec and rely on them getting called directly or through beg/end-of-defun?

Reply:
I think so, though it might be better to keep them interactive as long as obsoleted defaliases are in place.

Reply:
Alrighty, I renamed them and kept the interactive spec.